components/channel_header/channel_header.js

```diff
@@ -331,21 +331,27 @@ class ChannelHeader extends React.PureComponent {
         }
 
         if (isGroup) {
-            const nodes = [];
+            // map the displayname to the gm member users
+            const membersMap = {};
             for (const user of gmMembers) {
                 if (user.id === currentUser.id) {
                     continue;
                 }
                 const userDisplayName = Utils.getDisplayNameByUserId(user.id);
-                nodes.push((
-                    <React.Fragment key={userDisplayName}>
-                        {nodes.length !== 0 && ', '}
-                        {userDisplayName}
-                        <GuestBadge show={Utils.isGuest(user)}/>
-                    </React.Fragment>
-                ));
+                membersMap[userDisplayName] = user;
```

Comment:
what happens if two users have the same display name?

Reply:
It will replace it and only display the first one 😭 I would need the user id's for this approach to be reliable. 

The correct solution should be at the Redux level to get the list of group members sorted by the preference but not sure the effects of other callers to it. Thoughts @jespino?

Reply:
I'm not sure where the sorting happens here, but is it possible to take an array of users (or user IDs) and sort them by the display name with a custom compare function? You could do something like
```javascript
const displayNames = {};
for (const user of users) {
    displayNames[user.id] = Utils.getDisplayNameByUser(user);
}

const sortedUsers = [...users]; // Copying since .sort mutates the array
sortedUsers.sort((userA, userB) => displayNames[userA.id] < displayNames[userB.id]);
```
In that case, `displayNames` maps non-unique user IDs to display names, so it avoids that problem.

Reply:
Sorting of the display names is currently taking place at Redux level and. aside the display preference, it also takes into account the user locale. 

I've updated this implementation to account for duplicate display names by adding them in an array. Please review to see the latest changes @jespino @hmhealey

Reply:
Yes, nice solution. Thanks.